platform/ethereum/base.go

```diff
@@ -3,30 +3,37 @@ package ethereum
 import (
 	"github.com/trustwallet/blockatlas/coin"
 	"github.com/trustwallet/blockatlas/pkg/blockatlas"
+	"github.com/trustwallet/blockatlas/platform/ethereum/blockbook"
+	"github.com/trustwallet/blockatlas/platform/ethereum/collection"
+	"github.com/trustwallet/blockatlas/platform/ethereum/ens"
+	"github.com/trustwallet/blockatlas/platform/ethereum/trustray"
 )
 
 type Platform struct {
-	CoinIndex         uint
-	RpcURL            string
-	client            Client
-	collectionsClient CollectionsClient
-	ens               RpcClient
+	CoinIndex   uint
+	RpcURL      string
+	client      EthereumClient
+	collectible collection.Client
+	ens         ens.RpcClient
 }
 
-func Init(coin uint, api, rpc string) *Platform {
+func Init(coinType uint, api, rpc string) *Platform {
 	return &Platform{
-		CoinIndex: coin,
+		CoinIndex: coinType,
 		RpcURL:    rpc,
-		client:    Client{blockatlas.InitClient(api)},
-		ens:       RpcClient{blockatlas.InitJSONClient(rpc)},
+		ens:       ens.RpcClient{Request: blockatlas.InitJSONClient(rpc)},
+		client:    &trustray.Client{Request: blockatlas.InitClient(api)},
 	}
 }
 
-func InitWitCollection(coin uint, api, rpc, collectionApi, collectionKey string) *Platform {
-	p := Init(coin, api, rpc)
-	p.collectionsClient = CollectionsClient{blockatlas.InitClient(collectionApi)}
-	p.collectionsClient.Headers["X-API-KEY"] = collectionKey
-	return p
+func InitWitCollection(coinType uint, api, rpc, blockbookApi, collectionApi, collectionKey string) *Platform {
+	platform := Init(coinType, api, rpc)
+	if coinType == coin.ETH {
```

Comment:
can we improve this logic? blockbook only available on ETH node, why do we init collecitble here? if they are only ETH as well